packages/calcite-components/src/components/card-group/card-group.tsx

```diff
@@ -0,0 +1,251 @@
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /** Emits when the component's selection changes and the `selectionMode` is not `none`. */
+  @Event({ cancelable: false }) calciteCardGroupSelect: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback(): void {
+    connectInteractive(this);
+    this.mutationObserver?.observe(this.el, { childList: true, subtree: true });
+  }
+
+  componentDidRender(): void {
+    disconnectInteractive(this);
+    updateHostInteraction(this);
+  }
+
+  componentDidLoad(): void {
+    setComponentLoaded(this);
+  }
+
+  disconnectedCallback(): void {
+    this.mutationObserver?.disconnect();
+  }
+
+  async componentWillLoad(): Promise<void> {
+    setUpLoadableComponent(this);
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInternalCardKeyEvent")
+  calciteInternalCardKeyEventListener(event: CustomEvent): void {
+    if (event.composedPath().includes(this.el)) {
+      const interactiveItems = this.items.filter((el) => !el.disabled);
+      switch (event.detail.key) {
+        case "ArrowRight":
+          focusElementInGroup(interactiveItems, event.detail.target, "next");
+          break;
+        case "ArrowLeft":
+          focusElementInGroup(interactiveItems, event.detail.target, "previous");
+          break;
+        case "Home":
+          focusElementInGroup(interactiveItems, event.detail.target, "first");
+          break;
+        case "End":
+          focusElementInGroup(interactiveItems, event.detail.target, "last");
+          break;
+      }
+    }
+  }
+
+  @Listen("calciteCardSelect")
+  calciteCardSelectListener(event: CustomEvent): void {
+    if (
+      event.composedPath().includes(this.el) &&
+      !(event.target as HTMLCalciteCardElement).selectable
+    ) {
+      this.setSelectedItems(true, event.target as HTMLCalciteCardElement);
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Sets focus on the component's first focusable element.
+   */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
+    if (!this.disabled) {
+      (this.selectedItems[0] || this.items[0])?.setFocus();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private updateItems = (event?: Event): void => {
```

Comment:
Can you move the event argument to a separate event handler?

Reply:
@jcfranco I moved the slot change one into its own handler, can you check latest and see if it's ok?